src/server/pkg/hashtree/cache.go

```diff
@@ -137,13 +137,13 @@ func NewMergeCache(root string) *MergeCache {
 }
 
 // Put puts an id/hashtree pair in the cache and reads the hashtree from the passed in io.Reader.
-func (c *MergeCache) Put(id int64, tree io.Reader) (retErr error) {
-	return c.Cache.Put(fmt.Sprint(id), tree)
+func (c *MergeCache) Put(id string, tree io.Reader) (retErr error) {
```

Comment:
We no longer use datum indexes in hashtree chunks, so a string identifier is nice to have (which is great because it was a string in the underlying implementation anyway).

Reply:
I think we will be getting rid of this with the new storage layer. The model I am thinking we should move to (which I will need to flesh out more in a write up) is that we setup two file sets for a datum set (datum and output file set) and put files directly into those as datums are processed. The key negative implication of this is that machine / worker failures when processing a datum set result with the full datum set being lost somewhat (output upload may be deduped if garbage collection is infrequent). I think this makes sense because we want the datum set to be the granularity of a batch and we do not want to do an object storage upload/check for every datum. Skipping would operate on a streaming read of the previous datum file set, datums that change (same path but different content) would result with a delete then append on the base file set (technically compaction levels) for the compaction. I am going to need to read over these changes more to figure out if the parallel part of these changes might make the model I have in mind incompatible, but I think it is a simple way to be performant across workloads.

Reply:
I think the write up would be more so an update on https://github.com/pachyderm/pachyderm/issues/3703, than a full rewrite.